packages/components/managed-data-table/src/managed-data-table.js

```diff
@@ -0,0 +1,148 @@
+        onSortChange={props.onSortChange}
+        sortDirection={props.sortDirection}
+      />
+    </Spacings.Stack>
+  );
+};
+
+ManagedDataTable.propTypes = {
+  rows: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      /* defaults to `auto` */
+      width: PropTypes.string,
+      label: PropTypes.node.isRequired,
+      align: PropTypes.oneOf(['left', 'center', 'right']),
+      onClick: PropTypes.func,
+      /* custom item renderer, specific for items of this column */
+      renderItem: PropTypes.func,
+      isTruncated: PropTypes.bool,
+      isSortable: PropTypes.bool,
+      disableResizing: PropTypes.bool,
+      shouldIgnoreRowClick: PropTypes.bool,
+    })
+  ).isRequired,
+  footer: PropTypes.node,
+  maxWidth: PropTypes.number,
+  maxHeight: PropTypes.number,
+  onRowClick: PropTypes.func,
+  disableHeaderStickiness: PropTypes.bool,
+  disableFooterStickiness: PropTypes.bool,
+  /* the default item (cell) renderer.
+  an existing per-column `renderItem` func takes precedence over this */
+  itemRenderer: PropTypes.func.isRequired,
+  /* the default cell alignment
+  an existing per-column `align` property takes precedence over this */
+  isHeaderSticky: PropTypes.bool,
+  wrapHeaderLabels: PropTypes.bool,
+  verticalCellAlignment: PropTypes.oneOf(['top', 'center', 'bottom']),
+  horizontalCellAlignment: PropTypes.oneOf(['left', 'center', 'right']),
+  /* Sorting props: */
+  sortedBy: PropTypes.string,
+  onSortChange: PropTypes.func,
+  sortDirection: PropTypes.oneOf(['desc', 'asc']),
+  /* Table settings props */
+  displaySettings: PropTypes.shape({
+    disableDisplaySettings: PropTypes.bool,
+    isCondensed: PropTypes.bool,
+    isWrappingText: requiredIf(
+      PropTypes.bool,
+      (props) => props.displaySettings.disableDisplaySettings
+    ),
+    primaryButton: PropTypes.element,
+    secondaryButton: PropTypes.element,
+  }),
+  columnManager: PropTypes.shape({
+    disableColumnManager: PropTypes.bool,
+    visibleColumnKeys: requiredIf(
+      PropTypes.array,
+      (props) => !props.columnManager.disableColumnManager
+    ),
+    hideableColumns: requiredIf(
+      PropTypes.array,
+      (props) => !props.columnManager.disableColumnManager
+    ),
+    areHiddenColumnsSearchable: PropTypes.bool,
+    searchHiddenColumns: requiredIf(
+      PropTypes.func,
+      (props) =>
+        !props.columnManager.disableColumnManager &&
+        props.columnManager.areHiddenColumnsSearchable
+    ),
+    searchHiddenColumnsPlaceholder: PropTypes.string,
+    primaryButton: PropTypes.element,
+    secondaryButton: PropTypes.element,
+  }),
+  onSettingsChange: requiredIf(
+    PropTypes.func,
+    (props) =>
+      !props.displaySettings.disableColumnManager ||
+      !props.columnManager.disableDisplaySettings
+  ),
+  /* Top bar */
+  topBar: PropTypes.node,
```

Comment:
Is this supposed to be any React node, or only a string? It's not clear because in the tests and storybook use pass a string.

Reply:
Can be any React node. I'm updating the test so at least it's reflected there.